src/fontdata.cpp

```diff
@@ -74,3 +74,34 @@ void FontData::initFromStr(const string& source)
                 string("cannot allocate font \'") + source + "\'");
     }
 }
+
+/**
+ * @brief compute the with of the given text
+ * @param text The text
+ * @param glyphs Only consider that many glyphs from the given text
+ * @return The width in pixels
+ */
+int FontData::textwidth(const string& text, size_t len) const
+{
+    if (!s_xconnection) {
+        return 0;
+    }
+    if (xftFont_) {
+        XGlyphInfo info;
+        XftTextExtentsUtf8(s_xconnection->display(),
+                           xftFont_,
+                           const_cast<FcChar8*>(reinterpret_cast<const FcChar8*>(text.data())),
+                           len,
+                           &info);
+        return info.xOff;
+    }
+    if (xFontSet_) {
+        XRectangle logical;
+        Xutf8TextExtents(xFontSet_, text.c_str(), static_cast<int>(len), nullptr, &logical);
+        return logical.width;
+    }
+    if (xFontStruct_) {
+        return XTextWidth(xFontStruct_, text.c_str(), len);
+    }
+    return -1;
```

Comment:
-1 or 0? If there is no x connection you chose 0, in the else case it's -1.

Reply:
Thanks! this should be '0'. I've only changed it to -1 for debugging. :-)

Reply:
Thanks :)